adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -2135,6 +2163,41 @@ trait MultisampleGenomicRDD[T, U <: MultisampleGenomicRDD[T, U]] extends Genomic
   }
 }
 
+/**
+ * A trait describing a GenomicDataset that also supports the Spark SQL APIs.
+ */
+trait DatasetBoundGenomicDataset[T, U <: Product, V <: GenomicDataset[T, U, V]] extends GenomicDataset[T, U, V] {
+
+  /**
+   * Caches underlying dataset in memory.
+   *
+   * @return V cached GenomicDataset
+   */
+  override def cache(): V = {
+    transformDataset(_.cache())
+  }
+
+  /**
+   * Persists underlying dataset in memory.
```

Comment:
Persists underlying dataset in memory. → Persists underlying dataset in memory or disk.